src/target/riscv/riscv-013.c

```diff
@@ -499,6 +521,8 @@ static dmi_status_t dmi_scan(struct target *target, uint32_t *address_in,
 
 	assert(info->abits != 0);
 
+	address_out += target->dbgbase;
```

Comment:
I'd recommend to create another variable and not modify the input parameter value (detail, but helps code readability):

```suggestion
	const unint32_t address_with_dbgbase = address_out + target->dbgbase;
```